odxtools/utils.py

```diff
@@ -60,3 +65,33 @@ def is_short_name_path(test_val: str) -> bool:
     See also: ISO 22901 section 7.3.13.3
     """
     return _short_name_path_pattern.fullmatch(test_val) is not None
+
+
+@overload
+def resolve_snref(target_short_name: str,
+                  items: Iterable[OdxNamed],
+                  expected_type: None = None) -> Any:
+    """Properly resolve a short name reference"""
+    ...
+
+
+@overload
+def resolve_snref(target_short_name: str, items: Iterable[OdxNamed], expected_type: Type[T]) -> T:
+    ...
+
+
+def resolve_snref(target_short_name: str,
```

Comment:
No `@overload` here?

Reply:
this is the implementation. As far as I know, these cannot be marked `@overload` because their signature must suitable for any of the overloads...